src/main/java/com/thealgorithms/stacks/InfixToPrefix.java

```diff
@@ -0,0 +1,93 @@
+package com.thealgorithms.stacks;
+
+import java.util.Stack;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public final class InfixToPrefix {
+    private InfixToPrefix() {
+    }
+
+    /**
+     * Convert an infix expression to a prefix expression.
+     *
+     * @param infixExpression the infix expression to convert
+     * @return the prefix expression
+     * @throws IllegalArgumentException if the infix expression has unbalanced brackets
+     * @throws NullPointerException     if the infix expression is null
+     */
+    public static String infix2Prefix(String infixExpression) throws IllegalArgumentException {
+        // Check for unbalanced brackets
+        if (!BalancedBrackets.isBalanced(filterBrackets(infixExpression))) {
+            throw new IllegalArgumentException("invalid expression");
+        }
+        // Null input
+        if (infixExpression == null) {
+            throw new NullPointerException("null input");
+        }
+        // Empty string
+        if (infixExpression.isEmpty()) {
```

Comment:
And it would be better to carry out the following sequence of checks:

```
    if (infixExpression == null) {
        throw new NullPointerException("Input expression cannot be null.");
    }

    if (infixExpression.trim().isEmpty()) {
        return "";
    }

    if (!BalancedBrackets.isBalanced(filterBrackets(infixExpression))) {
        throw new IllegalArgumentException("Invalid expression: unbalanced brackets.");
    }
```